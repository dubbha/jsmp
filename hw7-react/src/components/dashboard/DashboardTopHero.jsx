import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const DashboardTopHero = ({ hero }) => (
  <Link to={`/hero-details/${hero.id}`} className="col-1-4">
    <div className="module hero">
      <h4>{hero.name}</h4>
    </div>
  </Link>
);

DashboardTopHero.propTypes = {
  hero: PropTypes.object.isRequired,
};

export default DashboardTopHero;
