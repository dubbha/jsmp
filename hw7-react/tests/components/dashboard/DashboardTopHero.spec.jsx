import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import DashboardTopHero from 'components/dashboard/DashboardTopHero';

describe('DashboardTopHero', () => {
  it('should render dashboard top hero element', () => {
    const props = {
      hero: {
        id: 666,
        name: 'Jesus',
      }
    };
    const wrapper = shallow(<DashboardTopHero {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
