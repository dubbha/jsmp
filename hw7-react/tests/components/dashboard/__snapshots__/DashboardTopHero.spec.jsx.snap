// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardTopHero should render dashboard top hero element 1`] = `
<Link
  className="col-1-4"
  replace={false}
  to="/hero-details/666"
>
  <div
    className="module hero"
  >
    <h4>
      Jesus
    </h4>
  </div>
</Link>
`;
